build/accelerate_launch.py

```diff
@@ -70,35 +69,33 @@ def main():
             if actions_type_map.get(key) == "_AppendAction":
                 for param_val in val:
                     accelerate_launch_args.extend([f"--{key}", str(param_val)])
+            elif (actions_type_map.get(key) == "_StoreTrueAction" and val) or (
+                actions_type_map.get(key) == "_StoreFalseAction" and not val
+            ):
+                accelerate_launch_args.append(f"--{key}")
             else:
                 accelerate_launch_args.append(f"--{key}")
                 # Only need to add key for params that aren't flags ie. --quiet
                 if actions_type_map.get(key) == "_StoreAction":
                     accelerate_launch_args.append(str(val))
 
-    if not accelerate_config.get("config_file"):
-        if json_configs.get("multi_gpu"):
-            # add FSDP config
-            fsdp_filepath = os.getenv(
+    if accelerate_config.get("num_processes"):
+        if accelerate_config.get("num_processes") > 1:
+            # Add FSDP config
+            fsdp_filepath = accelerate_config.get("config_file") or os.getenv(
                 "FSDP_DEFAULTS_FILE_PATH", "/app/accelerate_fsdp_defaults.yaml"
             )
             if os.path.exists(fsdp_filepath):
-                logging.info("Setting accelerate config file to: %s", fsdp_filepath)
-                accelerate_launch_args.append("--config_file")
-                accelerate_launch_args.append(fsdp_filepath)
-
-            # add num_processes to overwrite config file set one
-            if not accelerate_config.get("num_processes"):
-                num_gpus = torch.cuda.device_count()
-                if num_gpus > 1:
-                    logging.info("Setting accelerate num processes to %s", num_gpus)
-                    accelerate_launch_args.append("--num_processes")
-                    accelerate_launch_args.append(str(num_gpus))
-        else:
-            accelerate_launch_args.append("--num_processes")
-            accelerate_launch_args.append("1")
-
-    # add training_script
+                logging.info("Using accelerate config file: %s", fsdp_filepath)
+                accelerate_launch_args.extend(["--config_file", fsdp_filepath])
+
+    else:
+        logging.info(
+            "num_processes param was not passed in. So accelerate launch will use \
+                     default num_processes from config file"
+        )
```

Comment:
This logging message is not always true, this logging message should run if num_processes is not set and a config_file is passed in. Otherwise, num_processes is never set

Reply:
Because if no config_file is passed and num_processes isnt passed, this will go into the `accelerate launch` logic to determine number of processes which will use the torch.cuda_count().

maybe just simplify to:
```suggestion
        logging.warning(
            "`num_processes` param was not passed in. `accelerate launch` will use \
                    num_processes from config_file if provided or determine resources available"
        )
```
or maybe to start just
```suggestion
        logging.info(
            "`num_processes` param was not passed in.
        )
```